### PR TITLE
refactor(lane_change): create lane change info data

### PR DESCRIPTION
**Pull request by purewater0901**
## Description
Create `LaneChangeInfo` data to store necessary information for the lane change module. Using this data structure, we can decrease many arguments in the code.

## Tests performed

Scenario Test 1574/1587
[TIER IV INTERNAL LINK](https://evaluation.tier4.jp/evaluation/reports/ffec18f4-24a1-5be2-ace3-700a3d1ffe95?project_id=prd_jt)

## Effects on system behavior

No effects on system behavior

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/include/behavior_path_planner/utils/lane_change/lane_change_path.hpp [hunk 1]
```diff
@@ -30,19 +30,9 @@ using behavior_path_planner::TurnSignalInfo;
...
   ShiftedPath shifted_path{};
-  ShiftLine shift_line{};
-
-  // longitudinal acceleration applied on the prepare and lane-changing phase
-  LaneChangePhaseInfo longitudinal_acceleration{0.0, 0.0};
-
-  LaneChangePhaseInfo length{0.0, 0.0};
-  LaneChangePhaseInfo duration{0.0, 0.0};
   PathWithLaneId prev_path{};
+  LaneChangeInfo lane_change_info{};
```
Comment (zulfaqar-azmi-t4) [inline planning/behavior_path_planner/include/behavior_path_planner/utils/lane_change/lane_change_path.hpp:35]:
```suggestion
  LaneChangeInfo info{};
```
Naming it as `info` is sufficient in my opinion.

File: planning/behavior_path_planner/src/utils/lane_change/utils.cpp [hunk 2]
```diff
@@ -228,14 +228,18 @@ bool isPathInLanelets(
...
+  const PathWithLaneId & target_segment, const PathWithLaneId & target_lane_reference_path,
+  const std::vector<std::vector<int64_t>> & sorted_lane_ids)
 {
+  const auto & shift_line = lane_change_info.shift_line;
+  const auto & original_lanelets = lane_change_info.reference_lanelets;
+  const auto & target_lanelets = lane_change_info.target_lanelets;
+  const auto & terminal_lane_changing_velocity = lane_change_info.terminal_lane_changing_velocity;
+  const auto & longitudinal_acceleration = lane_change_info.longitudinal_acceleration;
+  const auto & lane_change_velocity = lane_change_info.velocity;
+  const auto & lane_change_length = lane_change_info.length;
```
Comment (zulfaqar-azmi-t4) [inline planning/behavior_path_planner/src/utils/lane_change/utils.cpp:241]:
```suggestion
  const auto terminal_lane_changing_velocity = lane_change_info.terminal_lane_changing_velocity;
  const auto longitudinal_acceleration = lane_change_info.longitudinal_acceleration;
  const auto lane_change_velocity = lane_change_info.velocity;
  const auto lane_change_length = lane_change_info.length;
```

Actually for plain types, normally `&` is not included.

File: planning/behavior_path_planner/src/utils/lane_change/utils.cpp [hunk 3]
```diff
@@ -825,10 +822,11 @@ std::vector<PoseWithVelocityStamped> convertToPredictedPath(
...
   const auto & path = lane_change_path.path;
-  const auto & prepare_acc = lane_change_path.longitudinal_acceleration.prepare;
-  const auto & lane_changing_acc = lane_change_path.longitudinal_acceleration.lane_changing;
-  const auto & duration = lane_change_path.duration.sum();
-  const auto & prepare_time = lane_change_path.duration.prepare;
+  const auto & prepare_acc = lane_change_path.lane_change_info.longitudinal_acceleration.prepare;
+  const auto & lane_changing_acc =
+    lane_change_path.lane_change_info.longitudinal_acceleration.lane_changing;
+  const auto & duration = lane_change_path.lane_change_info.duration.sum();
+  const auto & prepare_time = lane_change_path.lane_change_info.duration.prepare;
```
Comment (zulfaqar-azmi-t4) [inline planning/behavior_path_planner/src/utils/lane_change/utils.cpp:829]:
```suggestion
  const auto prepare_acc = lane_change_path.lane_change_info.longitudinal_acceleration.prepare;
  const auto lane_changing_acc =
    lane_change_path.lane_change_info.longitudinal_acceleration.lane_changing;
  const auto duration = lane_change_path.lane_change_info.duration.sum();
  const auto prepare_time = lane_change_path.lane_change_info.duration.prepare;
```

Plain type

File: planning/behavior_path_planner/include/behavior_path_planner/utils/lane_change/lane_change_path.hpp (hunk 1 above)
Comment (purewater0901) [reply planning/behavior_path_planner/include/behavior_path_planner/utils/lane_change/lane_change_path.hpp]: Done with https://github.com/autowarefoundation/autoware.universe/pull/4261/commits/fa829c65fa7391e1160ddca47544a890a40daaee

**Comment by purewater0901** [comment]
@zulfaqar-azmi-t4 I change the code based on your comments

**Review by zulfaqar-azmi-t4 — APPROVED** [review]
LGTM

**Pull request merged**